core/src/main/java/org/axonframework/eventsourcing/eventstore/jdbc/JdbcEventStorageEngine.java

```diff
@@ -548,13 +548,15 @@ protected PreparedStatement readEventData(Connection connection, TrackingToken l
         } else {
             gaps = Collections.emptyList();
         }
-        sql += "ORDER BY " + schema.globalIndexColumn() + " ASC";
+        sql += "ORDER BY " + schema.globalIndexColumn() + " ASC ";
+        sql += "LIMIT " + batchSize;
```

Comment:
Note that the LIMIT keyword is dependent on the database implementation and will not work in general.
See the suggested approach in #419